builds/otel.sh

```diff
@@ -47,11 +47,16 @@ if [ "$SKIP_OTEL_JAVA" != "true" ]; then
     cd ../..
 fi
 
+if [ -z "${GO_BIN}" ]; then
+    GO_BIN=/usr/local/go/bin/go
+fi
+ARCH=$($GO_BIN env GOARCH)
+
 cd submodules/opentelemetry-operations-collector/otelopscol
-ARCH=$(/usr/local/go/bin/go env GOARCH)
-GO_BIN=/usr/local/go/bin/go \
-BUILDARCH=$ARCH \
-TARGETARCH=$ARCH \
+
+BUILDARCH="$ARCH" \
```

Comment:
I'm curious about this update. At some point i had to install directly Go with a newer version in my local dev environment because `ARCH=$(/usr/local/go/bin/go env GOARCH)` was failing.

What is the goal/reasoning for this update ?

Reply:
> On my machine, Go versions are managed by VSCode and installed in different places.

Ahhh! Thanks for the clarification! That was my issue, VSCode was handling the different Go versions, so i installed one main version in that path to avoid the issue.